### ISSUE TITLE
Enhancement: Set Custom Endpoint Vision Models and Select Vision Model

### ISSUE THREAD

**Issue by Passerby1011** [danny-avila/LibreChat#2103]
### What happened?

I used the librechat.yaml provided by the project to establish an OpenRouter endpoint configuration as follows.
#
    - name: "OpenRouter"
      apiKey: "sk-***********-*******-*************************************************************************************"
      baseURL: "https://openrouter.ai/api/v1"
      models:
        default: ["mistralai/mixtral-8x7b-instruct"]
        fetch: true
      titleConvo: true
      titleModel: "gpt-3.5-turbo"
      summarize: false
      summaryModel: "gpt-3.5-turbo"
      forcePrompt: false
      modelDisplayLabel: "OpenRouter"
  #
When I use claude3, questions with images are forced to be sent to gpt-4-vision-preview, as shown below, the first message "hello" is replied to by claude3, the second message with an image is sent by liberchat to gpt-4-vision-preview, and the second message with an image is sent to gpt-4-vision-preview by liberchat. The second message with the image was sent by liberchat to gpt-4-vision-preview, with my OpenRouter logs below.

### Steps to Reproduce

Creating custom endpoints using profiles, with image questioning

### Screenshots

![20240315_091229_057_copy](https://github.com/danny-avila/LibreChat/assets/63802758/f8b5a92f-329f-4902-9a3e-5c1cc59af1f3)
![20240315_091442_936_copy](https://github.com/danny-avila/LibreChat/assets/63802758/af4c6884-3b7e-48f4-9423-e53a2ff08a78)
![20240315_091605_340_copy](https://github.com/danny-avila/LibreChat/assets/63802758/2a0a44be-57cc-4390-bfdd-be631e14999a)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Not supported yet, but will add it soon

**Comment by danny-avila** [comment]
Just realized this is a duplicate issue, closing in favor of the older one https://github.com/danny-avila/LibreChat/issues/2103

**Comment by avimar** [comment]
Closed in favor of which issue, that's linked here? It's a link to here.

TL;DR:

1) can we update the list of vision models, and have an easy way to automatically update the list?
2) In my tests, the cheapest good model for images is google/gemini-flash-1.5 and I would want that as my default
3) Either disable images on non-supported models, or, have it pass ONLY THE IMAGE to a vision model to "convert it".
4) Here's a prompt to get a vision model to fully describe a picture, so that it can be passed back to a regular model.



This was discussed on discord today - people testing new vision models in openrouter and but was silently sending not just the image but the entire conversation to openai/4o-mini when using openrouter.

I'd like the idea of automatically _converting_ the image by a chosen provider and the continuing in the chosen model if it dosen't support images, but I would want to know, e.g. a notice: "[This image was automatically converted to text by OpenAI 4o-mini]"


Here's all the models currently supporting vision in openrouter, and a dependency-free script to generate it:
https://gist.github.com/avimar/e8f39f59f345157482f73efaee529086

(resorted alphabetically for clarity:)

```
[
'anthropic/claude-3-haiku:beta',
'anthropic/claude-3-haiku',
'anthropic/claude-3-opus:beta',
'anthropic/claude-3-opus',
'anthropic/claude-3-sonnet:beta',
'anthropic/claude-3-sonnet',
'anthropic/claude-3.5-sonnet:beta',
'anthropic/claude-3.5-sonnet',
'google/gemini-flash-1.5-exp',
'google/gemini-flash-1.5',
'google/gemini-flash-8b-1.5-exp',
'google/gemini-pro-1.5-exp',
'google/gemini-pro-1.5',
'google/gemini-pro-vision',
'google/gemini-pro',
'meta-llama/llama-3.2-11b-vision-instruct:free',
'meta-llama/llama-3.2-11b-vision-instruct',
'meta-llama/llama-3.2-90b-vision-instruct',
'mistralai/pixtral-12b',
'openai/chatgpt-4o-latest',
'openai/gpt-4-turbo',
'openai/gpt-4-vision-preview'
'openai/gpt-4o-2024-05-13',
'openai/gpt-4o-2024-08-06',
'openai/gpt-4o-mini-2024-07-18',
'openai/gpt-4o-mini',
'openai/gpt-4o:extended',
'openai/gpt-4o',
'qwen/qwen-2-vl-72b-instruct',
'qwen/qwen-2-vl-7b-instruct',
'qwen/qwen-2.5-72b-instruct',
]
  ```




I have this prompt saved for "please describe this image" and it seems to be more thorough:

```
Please describe the image in detail, covering relevant aspects such as:

  For photographs, illustrations, or artwork:
  - The main subject(s) and their appearance, positioning, and actions
  - The setting, background, and any notable objects or elements
  - Colors, lighting, and overall mood or atmosphere
  - Any interesting details, textures, or patterns
  - The style, technique, or medium used (if discernible)
  
  For screenshots or images containing text:
  - The content and purpose of the text
  - The layout, formatting, and organization of the information
  - Any notable visual elements, such as logos, icons, or graphics
  - The overall context or message conveyed by the screenshot
  
  For graphs, charts, or data visualizations:
  - The type of graph or chart (e.g., bar graph, line chart, pie chart)
  - The variables being compared or analyzed
  - Any trends, patterns, or outliers in the data
  - The axis labels, scales, and units of measurement
  - The title, legend, and any additional context provided
  
  Be as specific and descriptive as possible while maintaining clarity and concision.
  ```
  
![long story](https://github.com/user-attachments/assets/0c6d8c99-5c1b-4fb7-8f7d-e32d778b9a32)
Simple prompt: "tell me everything about this image"
Got me this with google flash:

```
This is a two-panel meme.  The top panel shows a father and son sitting on a couch. The son is holding a thick book and asks his father, "Dad, why is this book so thick?"

The bottom panel shows the same father, now closer and smiling, and implying a more complex explanation, saying, "It's a long story."

The humor comes from the implied complexity and possibly embarrassing or lengthy explanation behind the thickness of the book, left to the viewer's imagination.  The meme plays on the common experience of parents having to explain complicated situations to their children.
```

and the longer prompt got me this (it's all accurate, but all simpler models can't explain the pun):

```
Here's a description of the image:

The image is a two-panel meme.  The top panel shows a father and son sitting on a light beige couch in a well-lit living room. The father, a dark-skinned man with a short, styled haircut, is wearing a blue shirt and jeans. He's sitting slightly behind his son, a dark-skinned boy with a similar haircut, who is holding an open book.  The boy is asking his father, "Dad, why is this book so thick?" via a speech bubble above his head.  The background is blurry but shows a neutral-toned living room with elements suggestive of comfort. The lighting is soft and natural, contributing to a calm atmosphere.

The bottom panel shows a close-up of the father from the top panel, again dark-skinned with a similar haircut.  His expression is one of a knowing smile. A speech bubble from him states, "It's a long story."  The background is a plain, blurred, light-colored wall.  The mood is somewhat humorous due to the implied joke.

The meme's humor lies in the juxtaposition of the son's simple question and the father's cryptic, suggestive response that hints at a complex, potentially embarrassing or funny, backstory behind the book's thickness. The use of similar shots of the father in both panels connects the meme's two parts. The overall style is a digitally created meme, likely using photographs combined with added text bubbles.  A watermark for @PunHubOnline is visible in the top right corner of the middle panel.
```

**Comment by danny-avila** [comment]
this is the one I meant to point to: https://github.com/danny-avila/LibreChat/issues/1634
